src/sync/app.cpp

```diff
@@ -0,0 +1,163 @@
+            return completion_block(nullptr, std::make_unique<error::ServiceError>(err));
+        } catch (const error::AppError& err) {
+            return completion_block(nullptr, std::make_unique<error::AppError>(err));
+        }
+
+        std::string profile_route = util::format("%1/auth/profile", m_base_route);
+        std::string bearer = util::format("Bearer %1", sync_user->access_token());
+
+        GenericNetworkTransport::get()->send_request_to_server({
+            Method::get,
+            profile_route,
+            m_request_timeout_ms,
+            {
+                { "Content-Type", "application/json;charset=utf-8" },
+                { "Accept", "application/json" },
+                { "Authorization", bearer}
+            },
+            std::string()
+        }, [&](const Response profile_response) {
+            // if there is a already an error code, pass the error upstream
+            if (response_code_is_fatal(profile_response.http_status_code)) {
+                return completion_block(nullptr, handle_error(profile_response));
+            }
+
+            json = nlohmann::json::parse(profile_response.body);
+
+            std::vector<SyncUserIdentity> identities;
+            auto identities_json = json["identities"];
+
+            for (size_t i = 0; i < identities_json.size(); i++)
+            {
+                auto identity_json = identities_json[i];
+                identities.push_back(SyncUserIdentity(identity_json["id"], identity_json["provider_type"]));
+            }
+
+            sync_user->update_identities(identities);
+
+            auto profile_data = json["data"];
+
+            sync_user->update_user_profile(std::make_shared<SyncUserProfile>(WRAP_JSON_OPT(profile_data, "name", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "email", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "picture_url", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "first_name", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "last_name", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "gender", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "birthday", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "min_age", std::string),
+                                                                             WRAP_JSON_OPT(profile_data, "max_age", std::string)));
+
+            return completion_block(sync_user, {});
+        });
+    };
+
+    std::map<std::string, std::string> headers = {
```

Comment:
Currently, the SDK's allow you to configure additional headers, but I guess that is something we can add later.

Reply:
👍